pkg/hackernews/client.go

```diff
@@ -60,25 +59,20 @@ func (c *client) GetItem(id int) (*model.Item, error) {
 func (c *client) performRequest(path, method string, result interface{}) error {
 	request, err := http.NewRequest(method, path, nil)
 	if err != nil {
-		return fmt.Errorf("Failed to creae http request object: %w", err)
+		return fmt.Errorf("Failed to create http request object: %w", err)
 	}
 
 	response, err := c.httpClient.Do(request)
 	if err != nil {
 		return fmt.Errorf("Unable to make http call: %w", err)
 	}
-	defer func(Body io.ReadCloser) {
-		err = Body.Close()
-	}(response.Body)
-	if err != nil {
-		return fmt.Errorf("Unable to close connection: %w", err)
-	}
+
 	if response.StatusCode < http.StatusOK || response.StatusCode >= http.StatusMultipleChoices {
 		return fmt.Errorf("Unexpected status code returned. Got %d status code", response.StatusCode)
 	}
 
 	if err := json.NewDecoder(response.Body).Decode(result); err != nil {
 		return fmt.Errorf("Unabled to decode response: %w", err)
 	}
-	return nil
+	return response.Body.Close()
```

Comment:
This should be called using a `defer` straight after the `c.httpClient.Do` `err != nil` check. If either of the two `if` statements above fail, then the function will return without cleaning up the request body. By using `defer` you ensure that the body is cleared if regardless of early returns or any `panic`s

Reply:
Fixed